src/core/Expander/ExpanderGroup/ExpanderGroup.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { Component, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { HtmlDiv, HtmlButton, HtmlButtonProps, HtmlSpan } from '../../../reset';
+import { baseStyles } from './ExpanderGroup.baseStyles';
+import { VisuallyHidden } from '../../../components';
+
+const baseClassName = 'fi-expander-group';
+const openClassName = `${baseClassName}--open`;
+const expandersContainerClassName = `${baseClassName}_expanders`;
+const openAllButtonClassName = `${baseClassName}_all-button`;
+interface InternalExpanderGroupProps {
+  /** Expanders and option other components */
+  children: ReactNode;
+  /** 'Open all' button text */
+  OpenAllText: string;
+  /** 'Close all'-component (Button) */
+  CloseAllText: string;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** 'Open all' button text for screen readers, hides OpenAllText for screen readers if provided */
+  AriaOpenAllText?: string;
+  /** 'Close all' button text for screen readers, hides CloseAllText for screen readers if provided */
+  AriaCloseAllText?: string;
+  /** Open/Close all button props */
+  toggleAllButtonProps?: Omit<
+    HtmlButtonProps,
+    | 'onClick'
+    | 'onMouseDown'
+    | 'onMouseUp'
+    | 'onKeyPress'
+    | 'onKeyUp'
+    | 'onKeyDown'
+  >;
+}
+
+interface ExpanderOpenStates {
+  [key: string]: boolean;
+}
+
+type ExpanderGroupTargetOpenState = {
+  targetOpenState: boolean;
+};
+
+interface ExpanderGroupState {
+  /** Expanders that are open */
+  expanders: ExpanderOpenStates;
+  /** State change transition request */
+  expanderGroupOpenState: ExpanderGroupTargetOpenState;
+}
+
+export interface ExpanderGroupProviderState {
+  onExpanderOpenChange: (id: string, toState: boolean | undefined) => void;
```

Comment:
Maybe using the `targetState` here also, instead of `toState`?

Reply:
Changed it, but it should actually be "newState" instead of toState. newState was already used in handleExpanderOpenChange function and reflects the use better in my opinion, because it's not a request to change the state, but rather just informing that the child expander state has already changed.